packages/devtools_app/lib/src/analytics/analytics.dart

```diff
@@ -167,24 +168,24 @@ class GtagExceptionDevTools extends GtagException {
   external String get flutter_client_id;
 }
 
-bool _gaEnabled;
+ValueNotifier<bool> gaEnabledNotifier = ValueNotifier(false);
 
 // Exposed function to JS via allowInterop.
-bool gaEnabled() => _gaEnabled;
+bool gaEnabled() => gaEnabledNotifier.value;
 
 /// Request DevTools property value 'enabled' (GA enabled) stored in the file
 /// '~\.devtools'.
 Future<bool> isAnalyticsEnabled() async {
```

Comment:
remove these two methods and trigger sending events to the server based on listening for changes to the gaEnabledNotifier. That way the notifier and the server can't get out of sync. Right now someone could modify gaEnabledNotifier and then the server wouldn't record the value.

Reply:
I took a different approach that accomplishes the same ends. I made the notifier private and added a public listenable getter. So the notifier value can only be modified in the context of these methods.